src/mame/drivers/williams.cpp

```diff
@@ -746,10 +746,10 @@ static INPUT_PORTS_START( defender )
 	PORT_BIT( 0xfe, IP_ACTIVE_HIGH, IPT_UNKNOWN )
 
 	PORT_START("IN2")
-	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_OTHER ) PORT_NAME("Auto Up / Manual Down") PORT_TOGGLE PORT_CODE(KEYCODE_F1)
-	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_OTHER ) PORT_NAME("Advance") PORT_CODE(KEYCODE_F2)
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_SERVICE4 ) PORT_NAME("Auto Up / Manual Down") PORT_TOGGLE
+	PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_SERVICE ) PORT_NAME("Advance")
 	PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_COIN3 )
-	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_OTHER ) PORT_NAME("High Score Reset") PORT_CODE(KEYCODE_9)
+	PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_SERVICE1 ) PORT_NAME("High Score Reset")
```

Comment:
Don't we have an "NVRAM reset" assignment that can be used for this, or is that already in use in this driver?  Same applies for the other systems.

Reply:
There is "IPT_MEMORY_RESET" with default key assignment "F1".
It doesn't appear to be specific to NVRAM, Though it is used as such.

However,
Previous "High Score Reset" was hard coded to key "9" (Typically IPT_SERVICE1)
Previous "Auto Up / Manual Down" was hard coded to key "F1" (Typically IPT_MEMORY_RESET)

Obviously if we want to use IPT_MEMORY_RESET then previous assignments should be reversed.
Suggested "High Score Reset" to be mapped to IPT_MEMORY_RESET (F1)
Suggested "Auto Up / Manual Down" to be mapped to IPT_SERVICE1 (9)

I will change to the suggested key mapping.